webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -105,31 +104,27 @@ const ProjectDetails = memo((projectDetails: ProjectDetails) => {
 
     // Reset the state when the component is unmounted
     useEffect(() => {
-        if (!getWebsocketEventState(state).isSubscriptionDeleted) {
-            fetchSubscriptionList();
-        }
-
         return () => {
             handleResetProjectDetails();
         };
     }, []);
 
     useEffect(() => {
-        if (subscriptionData) {
-            if (showAllSubscriptions) {
-                setSubscriptionList(subscriptionData);
-            } else {
-                setSubscriptionList(getCurrentChannelSubscriptions(subscriptionData, currentChannelId));
-            }
+        /**
+         * Prevent calling API to fetch subscription list twice on switching channel
+         *
+         * If current channel is changed and "showAllSubscription" was true on the last channel then
+         * this useEffect runs twice because "subscriptionListApiParams" is modified twice
+         * once when "currentChannelId" is updated and other time when "showAllSubscription" is set to false
+         */
+        if (showAllSubscription && previousState?.currentChannelId !== currentChannelId) {
+            return;
         }
-    }, [subscriptionData, showAllSubscriptions]);
+        fetchSubscriptionList();
+    }, [subscriptionListApiParams]);
 
-    // Update subscription list on switching channels
     useEffect(() => {
-        if (subscriptionData) {
-            setShowAllSubscriptions(false);
-            setSubscriptionList(getCurrentChannelSubscriptions(subscriptionData, currentChannelId));
-        }
+        setShowAllSubscriptions(false);
```

Comment:
Why are we toggling the "Show all subscriptions" switch if the channel is changed? We're not doing this in ServiceNow.

Reply:
I think we should discuss this with Mike also.